lib/bugsnag/configuration.rb

```diff
@@ -184,6 +180,16 @@ def debug(message)
       logger.debug(PROG_NAME) { message }
     end
 
+    ##
+    # Parses and sets proxy from a uri
+    def parse_proxy(uri)
+      proxy = URI.parse(uri)
```

Comment:
Would there be any problems if `uri` was nil at this point?

Reply:
Absolutely, however this would only occur if `parse_proxy` was called with `nil`, in which case `URI` would spit out a legible error i.e. `**URI::InvalidURIError: bad URI(is not URI?):**`

Reply:
I could catch and log that error, but I think failing loudly is more valuable in most cases.